packages/playback-core/src/renditions.ts

```diff
@@ -0,0 +1,103 @@
+import Hls from './hls';
+import type { HlsInterface } from './hls';
+import type { MediaTracks } from './types';
+import type { TrackEvent } from 'media-tracks';
+
+export function setupRenditions(customMediaEl: MediaTracks, hls: HlsInterface) {
+  if (!('videoTracks' in customMediaEl)) return;
+
+  // Create a map to save the unique id's we create for each level and rendition.
+  // hls.js uses the levels array index primarily but we'll use the id to have a
+  // 1 to 1 relation from rendition to level.
+  const levelIdMap = new WeakMap();
+
+  hls.on(Hls.Events.MANIFEST_PARSED, function (event, data) {
+    removeAllVideoTracks();
+
+    const videoTrack = customMediaEl.addVideoTrack('main');
+    videoTrack.selected = true;
+
+    for (const [id, level] of data.levels.entries()) {
+      const videoRendition = videoTrack.addRendition(
+        level.url[0],
+        level.width,
+        level.height,
+        level.videoCodec,
+        level.bitrate
+      );
+
+      // The returned levels all have an id of `0`, save the id in a WeakMap.
+      levelIdMap.set(level, `${id}`);
+
+      videoRendition.id = `${id}`;
+      videoRendition.enabled = hls.autoLevelEnabled;
```

Comment:
I believe this isn't quite right. enabled wrt the media-ui-extensions means it is selectable. autoLevelEnabled means the HLS ABR level switching is enabled.

Reply:
discussed internally, in this iteration `enabled` is used as a way to select 1, lock to 2, 3, or multiple renditions.

this will be discussed further in the future if there is a need for an API to select just 1 or a need for a flag to indicate the rendition is hidden / not possible to use.